### PR TITLE
Connection Register: Add current user email to connection register request

### PR DESCRIPTION
**Pull request by dereksmart**
#### Changes proposed in this Pull Request:

This adds the current user's email address to send as part of the register request body. The intent here is to be able to use this information in the .com side, by adding the register to the site's audit log. We can then use this email address, along with the jetpack_version and external user ID to log all required data for ToS acceptance.

#### Jetpack product discussion

Much of this is motivated by the ToS conversation, recapped here: p9dueE-1Gc-p2, but there may be more value to storing this data later on, such as back-checking who we expect the be the "site owner", a way to communicate to someone if the connection breaks, or to follow-up with them for other reasons. 

#### Does this pull request change what data or activity we track or use?

Yes. We were not sending this email address until now, nor were we storing it. 

#### Testing instructions:

- Disconnect Jetpack
- Sandbox your connection, and point to this patch D47562-code
- Click "Set up Jetpack" - this will register the site. 
- You do not need to complete the whole user auth.
- Check the site's Report Card audit log in wordpress.com 
- Verify the email was logged accurately. 

#### Proposed changelog entry for your changes:

N/A

### CODE REVIEW

File: packages/connection/src/class-manager.php [hunk 1]
```diff
@@ -861,6 +861,7 @@ public function register( $api_endpoint = 'register' ) {
 				'site_created'    => $this->get_assumed_site_creation_date(),
 				'jetpack_version' => Constants::get_constant( 'JETPACK__VERSION' ),
 				'ABSPATH'         => Constants::get_constant( 'ABSPATH' ),
+				'email'           => wp_get_current_user()->user_email,
```
Comment (ebinnion) [inline packages/connection/src/class-manager.php:864]: Nitpick - I'd prefer if we were more specific with the naming here. For example, `current_user_email`, `tos_accepted_email`, etc. I could imagine a case where we also want to send the administrator email address as well. Being more explicit will make it easier to distinguish on WP.com.

Comment (dereksmart) [reply packages/connection/src/class-manager.php:864]: Thanks, I was on the fence. Going with `current_user_email`

Comment (dereksmart) [reply packages/connection/src/class-manager.php]: done, and updated the phab revision

**Comment by jeherve** [comment]
Cherry-picked to `branch-8.8` in eba6fa2ec276a5c0ae904b318ae68a4675b2f21a

**Comment by jeherve** [comment]
Released a new version of the Connection package with that change, updated Terms Of Service, Tracking, JITM, Sync and ABTest with the updated dependency, and switched the release branch to that new version in a05b5e1766197f6e47109147b647c41b3c8410ea

**Pull request merged**